test/configure-errors.js

```diff
@@ -0,0 +1,107 @@
+(function() {
+  var loopback = require('loopback');
+  var app = loopback();
+  var m = require('./init');
+  var mock = require('mock-require');
+  var PassportConfigurator = require('../lib/passport-configurator.js');
+  var passportConfigurator = new PassportConfigurator(app);
+  var passport = require('passport');
+  var assert = require('assert');
+  var request = require('supertest');
+  var sinon = require('sinon');
+  var passportAuthenticate;
+
+  //mock a strategy to work with
+  function PassportFacebookStrategy(options, verify) {};
+  PassportFacebookStrategy.prototype.authenticate = function(req, options) {};
+  mock('passport-facebook', PassportFacebookStrategy);
+
+  describe('Login failure', function() {
+    before(function(done) {
+      var options = {
+        'provider': 'facebook',
+        'module': 'passport-facebook',
+        'callbackURL': '/auth/facebook/callback',
+        'authPath': '/auth/facebook',
+        'callbackPath': '/auth/facebook/callback',
+        'successRedirect': '/auth/facebook/redirect',
+        'failureRedirect': '/auth/facebook-failure',
+        'failureQueryString': true,
+        'link': false,
+      };
+      passportConfigurator.configureProvider('facebook-login', options);
+      done();
+    });
+
+    afterEach(function(done) {
+      if (passportAuthenticate) passportAuthenticate.restore();
+      done();
+    });
```

Comment:
move the aftereach above the actual tests

before() {}
afterEach() {}

it() {}

it() {}
```